### ISSUE TITLE
Change default database configuration by using an environment variable

### ISSUE THREAD

**Issue by V-ed** [V-ed/Automne2017CakePHPProject#3]
It would be great to change the default database for the whole application instead of overriding CakePHP's `defaultConnectionName()` on every single table available.

This would be great to load two configuration and switch between them easily (ex. having a *mysql* database but load the *sqlite* configuration on load).

Thing is, most plugins uses the `default` configuration and thus, cannot load side datasource configurations and the switch is made harder.

**Comment by V-ed** [comment]
Asked the teacher if he had a solution for this and after some research he sent me [this link](https://stackoverflow.com/questions/43126347/change-cakephp-database-configuration-globally).

I then followed by using the third way the one asking the question proposed and modified my `app.php` to my own needs :

```php
$config = [
    [...],
    'Datasources' => [
        'default' => [...],
        'sqlite' => [...],
        'test' => [...],
        'test_sqlite' => [...],
    ],
    [...],
];
```

https://github.com/V-ed/Automne2017CakePHPProject/blob/ac3a4fe483ea926f4ee0d8b58fb055b9b7c674b1/config/app.php#L422-L431

Basically, I always use the default connection but override it with the data contained in the environment variable `DATABASE_SOURCE` (with CakePHP's **test_*[database_name]*** test handling).
